python-avd/pyavd/_eos_designs/structured_config/underlay/prefix_lists.py

```diff
@@ -93,34 +83,27 @@ def prefix_lists(self: AvdStructuredConfigUnderlayProtocol) -> list | None:
                 ):
                     sequence_number += 10
                     subnet = str(ip_network(f"{uplink['ip_address']}/{uplink['prefix_length']}", strict=False))
-                    p2p_links_sequence_numbers.append({"sequence": sequence_number, "action": f"permit {subnet}"})
+                    p2p_links_sequence_numbers.append_new(sequence=sequence_number, action=f"permit {subnet}")
         if p2p_links_sequence_numbers:
-            prefix_lists.append({"name": "PL-P2P-LINKS", "sequence_numbers": p2p_links_sequence_numbers})
-
-        return prefix_lists
+            self.structured_config.prefix_lists.append_new(name="PL-P2P-LINKS", sequence_numbers=p2p_links_sequence_numbers)
 
-    @cached_property
-    def ipv6_prefix_lists(self: AvdStructuredConfigUnderlayProtocol) -> list | None:
-        """Return structured config for IPv6 prefix_lists."""
+    @structured_config_contributor
+    def ipv6_prefix_lists(self: AvdStructuredConfigUnderlayProtocol) -> None:
+        """Set the structured config for IPv6 prefix_lists."""
         if self.shared_utils.underlay_bgp is not True:
-            return None
+            return
 
         if self.shared_utils.underlay_ipv6 is not True:
-            return None
+            return
 
         if self.shared_utils.overlay_routing_protocol == "none" and not self.shared_utils.is_wan_router:
-            return None
+            return
 
         if not self.inputs.underlay_filter_redistribute_connected:
-            return None
+            return
```

Comment:
would make sense to hamonise this with the ip_prefix_list refactor

Reply:
Fixed!